app/actions/websocket/channel.test.ts

```diff
@@ -0,0 +1,577 @@
+import {deleteChannelMembership, getChannelById, getCurrentChannel, prepareMyChannelsForTeam} from '@queries/servers/channel';
+import {setCurrentTeamId, getCurrentChannelId, getCurrentTeamId, getConfig} from '@queries/servers/system';
+import {getCurrentUser, getTeammateNameDisplay, getUserById} from '@queries/servers/user';
+import EphemeralStore from '@store/ephemeral_store';
+
+import {handleChannelCreatedEvent, handleChannelUnarchiveEvent, handleChannelConvertedEvent, handleChannelUpdatedEvent, handleChannelViewedEvent, handleMultipleChannelsViewedEvent, handleChannelMemberUpdatedEvent, handleChannelDeletedEvent, handleDirectAddedEvent, handleUserAddedToChannelEvent, handleUserRemovedFromChannelEvent} from './channel';
+
+import type ServerDataOperator from '@database/operator/server_data_operator';
+
+jest.mock('@database/manager');
+jest.mock('@store/ephemeral_store');
+jest.mock('@actions/local/category');
+jest.mock('@actions/remote/channel');
+jest.mock('@actions/remote/role');
+jest.mock('@actions/remote/post');
+jest.mock('@actions/remote/user');
+jest.mock('@actions/local/channel');
+jest.mock('@actions/local/post');
+jest.mock('@queries/servers/channel');
+jest.mock('@queries/servers/system');
+jest.mock('@queries/servers/user');
+jest.mock('@calls/actions');
+jest.mock('@actions/local/user');
+jest.mock('@utils/log');
+jest.mock('@calls/state');
+jest.mock('@calls/actions/calls');
+
+const serverUrl = 'baseHandler.test.com';
+
+describe('WebSocket Channel Actions', () => {
+    let msg: any;
+    let operator: ServerDataOperator;
+
+    const teamId = 'teamid1';
+    const channelId = 'channelid1';
+    const userId = 'userid1';
+
+    beforeEach(async () => {
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+        msg = {
+            data: {
+                team_id: teamId,
+                channel_id: channelId,
+                channel: JSON.stringify({id: channelId, type: General.PRIVATE_CHANNEL, team_id: teamId}),
+            },
+            broadcast: {
+                channel_id: channelId,
+                userId,
+                user_id: userId,
+            },
+        };
+        DatabaseManager.getActiveServerUrl = jest.fn().mockResolvedValue(serverUrl);
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+        jest.restoreAllMocks();
+    });
+
+    describe('handleChannelCreatedEvent', () => {
+        it('should handle channel created event', async () => {
+            (EphemeralStore.creatingChannel as boolean) = false;
+            (getChannelById as jest.Mock).mockResolvedValue(null);
+            (fetchMyChannel as jest.Mock).mockResolvedValue({channels: [{}], memberships: [{}]});
+            (prepareMyChannelsForTeam as jest.Mock).mockResolvedValue([[]]);
+            (addChannelToDefaultCategory as jest.Mock).mockResolvedValue({models: []});
```

Comment:
I am not a great fan of such heavy mocking for two main reasons:
- The data is not real
- Changes beyond this file may not make the test break (integration testing vs pure unit testing)

It is a more philosophical discussion, so I am 0/5 on whether this require any change.

Reply:
>The data is not real

The data isn't real in any test. Even if it's more of an integration test somewhere there needs to be mock/fake data. We could make the data more realistic for these tests but it's not required by the implementation of the function so I don't see the point.

>Changes beyond this file may not make the test break (integration testing vs pure unit testing)

These should be unit tests imo and only focused on testing the functionality in the related functions for the most part. The unit tests of other files should cover their functionality. Then we have E2E tests to cover integration. I feel that having every test be an integration test would be overkill and make the tests too fragile but I'm open to having my mind changed.

Thoughts @enahum ?

Reply:
Continued the conversation here https://hub.mattermost.com/private-core/pl/dqauz7iohbdquratstmojgtyer